File: backend/internal/model/quotation.go
package model

import "time"

// Quotation represents a price quotation for steel materials.
type Quotation struct {
	ID               uint      `gorm:"column:id;primaryKey" json:"id"`
	UserID           uint      `gorm:"column:user_id" json:"user_id"`
	Title            string    `gorm:"column:title" json:"title"`
	CustomerName     string    `gorm:"column:customer_name" json:"customer_name"`
	Category         string    `gorm:"column:category" json:"category"`
	Spec             string    `gorm:"column:spec" json:"spec"`
	Quantity         float64   `gorm:"column:quantity" json:"quantity"`
	Unit             string    `gorm:"column:unit" json:"unit"`
	DeliveryLocation string    `gorm:"column:delivery_location" json:"delivery_location"`
	MaterialCost     float64   `gorm:"column:material_cost" json:"material_cost"`
	ProcessCost      float64   `gorm:"column:process_cost" json:"process_cost"`
	FreightCost      float64   `gorm:"column:freight_cost" json:"freight_cost"`
	TaxCost          float64   `gorm:"column:tax_cost" json:"tax_cost"`
	TotalPrice       float64   `gorm:"column:total_price" json:"total_price"`
	Status           string    `gorm:"column:status" json:"status"`
	CreatedAt        time.Time `gorm:"column:created_at" json:"created_at"`
	UpdatedAt        time.Time `gorm:"column:updated_at" json:"updated_at"`
}

// TableName returns the database table name for Quotation.
func (Quotation) TableName() string {
	return "quotations"
}

File: steel-agent-web/src/app/types/quotation.ts
// ============================================================
// 报价单相关 TypeScript 类型定义
// 对应后端 /api/v1/quotations/* 接口
// ============================================================

/** 报价单 */
export interface Quotation {
  id: number;
  title: string;
  user_id?: number;
  customer_name: string;
  category: string;
  spec: string;
  quantity: number;
  unit: string;
  material_cost: number;
  process_cost: number;
  freight_cost: number;
  tax_cost: number;
  total_price: number;
  status: 'draft' | 'sent' | 'accepted' | 'rejected';
  delivery_location: string;
  created_at: string;
  updated_at?: string;
}
